src/poc/miner_poc_grpc_client_statem.erl

```diff
@@ -611,13 +614,17 @@ do_send_report(Req, ReportType, Report, OnionKeyHash, Connection, RetryAttempts)
                 {ok, _} ->
                     ok;
                 _ ->
-                    ?MODULE:send_report(ReportType, Report, OnionKeyHash, RetryAttempts - 1)
+                    retry_report({send_report, ReportType, Report, OnionKeyHash, RetryAttempts - 1})
```

Comment:
could we add some failure logging here too? useful for debug and we can turn it down later.